sci-electronics/systemc/systemc-2.3.4-r1.ebuild

```diff
@@ -41,12 +40,27 @@ src_configure() {
 		--with-unix-layout
 }
 
+src_install() {
+	default
+	if use doc; then
+		if use examples; then
+			docompress -x /usr/share/doc/"${PF}"/examples
+		else
+			rm -r "${ED}"/usr/share/doc/"${PF}"/examples
+		fi
+	else
+		rm -r "${ED}"/usr/share/doc/"${PF}"
+	fi
+}
+
 pkg_postinst() {
-	elog "If you want to run the examples, you need to :"
-	elog "    tar xvfz ${PORTAGE_ACTUAL_DISTDIR}/${A}"
-	elog "    cd ${PN}-${MY_PV}"
-	elog "    mkdir build && cd build"
-	elog "    cmake .."
-	elog "    cd examples"
-	elog "    make check"
+	if use examples; then
+		elog "If you want to run the examples, you need to :"
+		elog "    tar xvfz ${PORTAGE_ACTUAL_DISTDIR}/${A}"
+		elog "    cd ${PN}-${MY_PV}"
+		elog "    mkdir build && cd build"
+		elog "    cmake .."
+		elog "    cd examples"
+		elog "    make check"
+	fi
 }
```

Comment:
is it possible to drop the whole pkg_postinst()? instead, let's build & install it in compile()/install() phase?

Reply:
> is it possible to drop the whole pkg_postinst()? instead, let's build & install it in compile()/install() phase?

Done, pkg_postinst() have been dropped. All the example will compile in test phase (with test use flag).